src/test/ui/rust-2018/uniform-paths/block-scoped-shadow-nested.rs

```diff
@@ -0,0 +1,22 @@
+// edition:2018
+
+#![feature(uniform_paths)]
+
+mod my {
+    pub mod sub {
+        pub fn bar() {}
+    }
+}
+
+mod sub {
+    pub fn bar() {}
+}
+
+fn foo() {
+    use my::sub;
+    {
+        use sub::bar; //~ ERROR `sub` is ambiguous
```

Comment:
Is the rule that, for `use sub::foo`, `sub` can refer to modules that were imported, but only if they were imported in some outer scope? I am comparing this test with [issue-54390.rs](https://github.com/rust-lang/rust/blob/346a2a9dc835ac0bd5589674d993b0e421f55c7f/src/test/ui/rust-2018/uniform-paths/issue-54390.rs), which gives an error in a very similar situation:

```rust
use std::fmt;
use fmt::Write;
fn main() {}
```

Reply:
`issue-54390.rs` gives a feature error because `#[feature(uniform_paths)]` is not enabled there, otherwise it's legal.

So, there's no any kind of "strictly outer scope" rule, both import and its source can be from the same module.